src/storage/src/types/sources.proto

```diff
@@ -201,6 +202,12 @@ message ProtoPubNubSourceConnection {
     string channel = 2;
 }
 
+message ProtoLoadGeneratorSourceConnection {
+    oneof generator {
+        google.protobuf.Empty counter = 1;
```

Comment:
Not thrilled about forcing ourselves to add a protobuf entry for each generator instead of using the string, but I found this overall better than using the string and having to have a to string and from string impls.